frontend/src/components/Order/components/PostcardAndComment/PostcardAndComment.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import styles from './PostcardAndComment.module.scss';
+import Button from '../../../ui-kit/components/Button';
+
+const PostcardAndComment = ({ postcardData, setDataOnSubmit }) => {
+  const [showPostcardText, setShowPostcardText] = useState(
+    postcardData?.postcard ? postcardData.postcard === 'Add postcard' : true,
+  );
+  const [showComment, setShowComment] = useState(postcardData?.comment !== '');
+
+  const { register, handleSubmit } = useForm({
+    defaultValues: {
+      postcard: postcardData?.postcard ? postcardData.postcard : 'Add postcard',
+      postcardText: postcardData?.postcardText,
+      comment: postcardData?.comment,
+    },
+  });
+
+  const onSubmit = (data) => {
+    let dataForOutput;
+    if (data.postcard === 'Add postcard') {
+      dataForOutput = 'Додати листівку';
+    } else {
+      dataForOutput = ' Без листівки';
+    }
+
+    setDataOnSubmit({
+      ...data,
+      outputString: dataForOutput,
+    });
+  };
+  return (
+    <form className={styles.postcardForm} onSubmit={handleSubmit(onSubmit)}>
+      <label htmlFor="postcard" className={styles.radioBtn}>
+        <input
+          type="radio"
+          id="postcard"
+          value="Add postcard"
+          {...register('postcard')}
+          onChange={() => {
+            setShowPostcardText(true);
+          }}
+        />
+        Додати листівку з побажанням
+      </label>
+      {showPostcardText && (
+        <textarea
+          placeholder="Напишіть ваше побажання"
+          {...register('postcardText')}
+        ></textarea>
+      )}
+      <label htmlFor="noPostcard" className={styles.radioBtn}>
+        <input
+          type="radio"
+          id="noPostcard"
+          value="No postcard"
+          {...register('postcard')}
+          onChange={() => {
+            setShowPostcardText(false);
+          }}
+        />
+        Без листівки
+      </label>
+      <div
+        className={`${styles.comment} ${showComment ? styles.commentOpen : ''}`}
+      >
+        <label htmlFor="comment" onClick={() => setShowComment(true)}>
+          Коментар до замовлення
```

Comment:
Are you sure it should be label? The comment field isn't hidden by default. And it doesn't hide comment field on click

Reply:
This text is associated to the textarea below. So I used the label.
Comment field is already corrected.